css/how_to_use_css_custom_properties.md

```diff
@@ -0,0 +1,154 @@
+
+## When to use CCP?
+
+The power of CCP is that it can be used in instances of the component.
+
+Let's say we have a general `.button` component and its default color is `red`
+
+We can use the same component and change the color by just declaring the variable in the component or its parent
+
+```HTML
+<!-- Simple HTML -->
+<div class="component">
+    <button class="button">...</button>
+</div>
+
+<div class="component-2">
+    <button class="button">...</button>
+</div>
+
+<div class="component-3">
+    <button class="button">...</button>
+</div>
+```
+
+```SCSS
+// General CSS
+:root {
+    --primary-color: red;
+}
+
+.button {
+    color: var(--primary-color);
+}
+```
+
+```SCSS
+// Instanced components
+.component-2 {
+    --primary-color: blue;
+}
+
+.component-3 {
+    --primary-color: green;
+}
+```
+
+[Simple example](https://codepen.io/eduardoallegrini/pen/WNvgWMp)
+
+## How to use CCP?
+
+From the previous example we can main form of manipulate the CCP is through CSS itself.
+
+This method can be extended to the theme or section following the Cascade rule of CSS
+
+### By CSS
+
+```HTML
+<!-- Advanced HTML -->
+<body>
+    <section class="section-1">
+        <div class="component-1">
+            <button class="button">...</button>
+        </div>
+
+        <div class="component-1">
+            <button class="button">...</button>
+        </div>
+
+        <div class="component-1">
+            <button class="button">...</button>
+        </div>
+    </section>
+
+    <section class="section-2">
+        <div class="component-1">
+            <button class="button">...</button>
+        </div>
+
+        <div class="component-1">
+            <button class="button">...</button>
+        </div>
+
+        <div class="component-1">
+            <button class="button">...</button>
+        </div>
+    </section>
+</body>
+```
+
+```SCSS
+// General CSS
+:root {
+    --primary-color: red;
+}
+
+.button {
+    color: var(--primary-color);
+}
+
+// Instanced components
+body {
+    --primary-color: blue;
+
+	.section-2 {
+		--primary-color: green;
+	}
+
+	.section-3 {
+		--primary-color: purple;
+
+		.component-3 {
+			--primary-color: orange;
+		}
+	}
+}
+```
+
+[Advanced example](https://codepen.io/eduardoallegrini/pen/gOpdyKP)
+
+### By JS
+
+Another method to use CCP in your application is with JS
+
+The `element.style.getPropertyValue("-variable");` allow us to get the custom-property of the element, if the element doesn’t have a defined custom-property, the function will return an empty argument
+
+```JS
+// get variable from inline style
+element.style.getPropertyValue("--my-var");
+```
+
+The `getComputedStyle(element).getPropertyValue("--variable");` allow us to get the custom-property of the element, if the element doesn’t have a defined custom-property, the function will look for it at the parent elements or at the root of our markup
```

Comment:
Drop initial "The", add trailing period.

You have a run-on sentence here that should be broken into two too:

> `getComputedStyle(element).getPropertyValue("--variable");` allow us to get the custom-property of the element. If the element doesn’t have a defined custom-property, the function will look for it at the parent elements or at the root of our markup.